Removes PSGallery availability checks

OCram85 opened the pull request:
## :package:  Changes
- removes `Get-PSGalleryApiAvailability` references and the function itself
- removes `Check-PSGalleryApiAvailability` references and the function itself
- removes unused variables
- Removes Pester test cases for
  - `Get-PSGalleryApiAvailability`
  - `Check-PSGalleryApiAvailability`

### :notebook_with_decorative_cover: General Changes
- Introduced vscode workspace settings to enable common style
  - VSCode settings contains defaults for the editor and PowerShell extension.
  - Tried to adopt [PowerShellPracticeAndStyle](https://github.com/PoshCode/PowerShellPracticeAndStyle)

### :bookmark:  Ref
- fixes #369

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/PowerShell/PowerShellGet?pullRequest=374) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/PowerShell/PowerShellGet?pullRequest=374) before we can accept your contribution.<br/><br/>:x: OCram85 [sign now](https://cla.opensource.microsoft.com/PowerShell/PowerShellGet?pullRequest=374)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/PowerShell/PowerShellGet?pullRequest=374) it.</sub>

edyoung commented:
Mi Marco, thanks for the PR. I don't see a problem, and expect to merge it once we have fixed the CI tests, which are currently failing for an unrelated reason. For future reference, in general I find it easier to review if a functional change is not 'mixed in' with reformatting the affected files, because it's trickier to spot the actual changes among all the style updates.

alerickson commented:
@OCram85 there's a missing `}` that should be on line 204 of PSGetModuleSource.Tests.ps1 (closing out the describe block on line 87), but everything else looks good

OCram85 commented:
@alerickson: Thanks for the help :D -  I've just updated the PR.

@edyoung: I'm not sure if I messed with the common style in this project. I personally like adding defaults with vscode workspace settings. I needed - I would also split this PR and introduce the defaults with a seperate PR which also would include fixing the style issues in all files ^^
